sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,645 @@
+ * @param[in]  ep       Remote endpoint of the session
+ * @param[out] remote   The created session, cannot be NULL
+ *
+ * @return  0 on success
+ * @return  -EAGAIN, if DTLS_HANDSHAKE_TIMEOUT is `0` and no data is available.
+ * @return  -EADDRNOTAVAIL, if the local endpoint of @p sock is not set.
+ * @return  -EINVAL, if @p remote is invalid or @p sock is not properly
+ *          initialized (or closed while sock_udp_recv() blocks).
+ * @return  -ENOBUFS, if buffer space is not large enough to store received
+ *          credentials.
+ * @return  -ETIMEDOUT, if timed out when trying to create session.
+ */
+int sock_dtls_session_create(sock_dtls_t *sock, sock_udp_ep_t *ep,
+                             sock_dtls_session_t *remote);
+
+/**
+ * @brief Destroys an existing DTLS session
+ *
+ * @pre `(sock != NULL) && (ep != NULL)`
+ *
+ * @param[in] sock      @ref sock_dtls_t, which the session is created on
+ * @param[in] remote    Remote session to destroy
+ */
+void sock_dtls_session_destroy(sock_dtls_t *sock, sock_dtls_session_t *remote);
+
+/**
+ * @brief Decrypts and reads a message from a remote peer.
+ *
+ * @param[in] sock      DTLS sock to use.
+ * @param[out] remote   Remote DTLS session of the received data.
+ *                      Cannot be NULL.
+ * @param[out] data     Pointer where the received data should be stored.
+ * @param[in] maxlen    Maximum space available at @p data.
+ * @param[in] timeout   Receive timeout in microseconds.
+ *                      If 0 and no data is available, the function returns
+ *                      immediately.
+ *                      May be SOCK_NO_TIMEOUT to wait until data
+ *                      is available.
+ *
+ * @note Function may block if data is not available and @p timeout != 0
+ *
+ * @return The number of bytes received on success
+ * @return  -EADDRNOTAVAIL, if the local endpoint of @p sock is not set.
+ * @return  -EAGAIN, if @p timeout is `0` and no data is available.
+ * @return  -EINVAL, if @p remote is invalid or @p sock is not properly
+ *          initialized (or closed while sock_dtls_recv() blocks).
+ * @return  -ENOBUFS, if buffer space is not large enough to store received
+ *          data.
+ * @return  -ENOMEM, if no memory was available to receive @p data.
+ * @return  -ETIMEDOUT, if @p timeout expired.
+ */
+ssize_t sock_dtls_recv(sock_dtls_t *sock, sock_dtls_session_t *remote,
+                       void *data, size_t maxlen, uint32_t timeout);
+
+/**
+ * @brief Encrypts and sends a message to a remote peer
+ *
+ * @param[in] sock      DTLS sock to use
+ * @param[in] remote    DTLS session to use. A new session will be created
+ *                      if no session exist between client and server.
+ * @param[in] data      Pointer where the data to be send are stored
+ * @param[in] len       Length of @p data to be send
+ *
+ * @note Function may block until a session is created if there is no
+ *       existing session with @p remote.
```

Comment:
Does this mean, `remote` can change? If so, it should be marked as `@param[in,out]`.

Reply:
No, `remote` does not change. The function only uses the information in `remote ` to create a session with the endpoint pointed to by `remote`.

Reply:
Ok